internal/bootstrap/options.go

```diff
@@ -112,3 +112,24 @@ func (o loggerOption) applyGit(b *PlainGitBootstrapper) {
 func (o loggerOption) applyGitProvider(b *GitProviderBootstrapper) {
 	b.logger = o.logger
 }
+
+func WithGitCommitSigning(path, passphrase string) Option {
```

Comment:
I think the `WithGitCommitSigning` (or `WithCommitSigning`) should be reserved for the most UX-friendly approach (trying to determine the user's agent), and this advanced option should be named something like: `WithCommitSigningKeyRing()` (or something else that properly explains what it entails).